modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/MuleApplicationPolicyProvider.java

```diff
@@ -105,6 +111,17 @@ public synchronized void addPolicy(PolicyTemplateDescriptor policyTemplateDescri
     }
   }
 
+  private boolean isPolicyReorder(RegisteredPolicyInstanceProvider provider, PolicyTemplateDescriptor policyTemplateDescriptor,
+                                  PolicyParametrization parametrization) {
+    return parametrization.getParameters().getOrDefault("isPolicyReorder", "false").equalsIgnoreCase("true");
+  }
+
+  private void reorderPolicy(RegisteredPolicyInstanceProvider provider, PolicyParametrization parametrization) {
+    provider.updateOrder(parametrization.getOrder());
+    registeredPolicyInstanceProviders.sort(null);
```

Comment:
extract this and other similar codes to its own method, so that comments can be added to it explaining why a list is used instead of a SortedMap.